src/stcal/ramp_fitting/ramp_fit.py

```diff
@@ -259,27 +260,30 @@ def suppress_one_group_saturated_ramps(ramp_data):
     dq = ramp_data.groupdq
     nints, ngroups, nrows, ncols = dq.shape
     sat_flag = ramp_data.flags_saturated
+    jump_flag = ramp_data.flags_jump_det
 
     ramp_data.one_groups = [None] * nints
 
     for integ in range(nints):
         ramp_data.one_groups[integ] = []
         intdq = dq[integ, :, :, :]
-
-        # Find ramps with a good zeroeth group, but saturated in
-        # the remainder of the ramp.
-        wh_one = groups_saturated_in_integration(intdq, sat_flag, ngroups - 1)
+        # Find ramps with only one group that is not saturated and
+        # not jump (ie only one good group)
+        bad_flags = np.bitwise_or(sat_flag, jump_flag)
+        bad_groups = np.zeros(intdq.shape, dtype=int)
+        bad_groups[np.where(np.bitwise_and(intdq, bad_flags))] = 1
+        nbad_groups = bad_groups.sum(axis=0)
+        wh_one = np.where(nbad_groups == (ngroups - 1))
 
         wh1_rows = wh_one[0]
         wh1_cols = wh_one[1]
         for n in range(len(wh1_rows)):
             row = wh1_rows[n]
             col = wh1_cols[n]
-            # For ramps that have good 0th group, but the rest of the
-            # ramp saturated, mark the 0th groups as saturated, too.
-
-            if ramp_data.groupdq[integ, 0, row, col] == 0:
-                ramp_data.groupdq[integ, 0, row, col] = sat_flag
+            # For ramps that have only one good group, mark that group as saturated
```

Comment:
i am not super familiar with ramp fitting and am trying to understand this. what is happening here - if there is one good group of data why is the 'saturation' flag being applied? saturation runs before this and if this group wasn't flagged then why is it being called saturated here?

i assume theyre being flagged so a ramp wont be fit? if thats the case why not the donotuse flag or something?

Reply:
This is happening during suppression of one group ramps.  There is an option to suppress ramps that saturate in group 1, but still has a good group 0.  To suppress this group 0, its DQ flag is set to saturation.